### PR TITLE
[ADD][14.0] New module hr_holidays_validator_list

### PR DESCRIPTION
**Pull request by victorvermot**
A module making replacing the current leave_manager_id field by a many2many field called leave_manager_ids. Thus allowing a user to have several leave managers.

TODO: 
- [ ] More unit tests

### CODE REVIEW

File: hr_holidays_validator_list/models/hr_employee.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        "res.users",
+        string="Time Off Managers",
+        store=True,
+        readonly=False,
+        help="""Select the users responsible for approving 'Time Off' of this employee.\n
+        If empty, the approval is done by an Administrator
+        or Approver (determined in settings/users).""",
+    )
+
+    def _set_leave_manager_id_from_leave_manager_ids(self, values):
```
Comment (StephaneMangin) [inline hr_holidays_validator_list/models/hr_employee.py]: You should have used a stored readable only computed method on leave_manager_id with a depends on leave_manager_ids to propagate any change.

File: hr_holidays_validator_list/models/hr_employee.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        if "leave_manager_ids" in values:
+            if not len(values["leave_manager_ids"][0][-1]):
+                values["leave_manager_id"] = None
+            elif "leave_manager_ids" in values.keys():
+                values["leave_manager_id"] = values["leave_manager_ids"][0][-1][-1]
+        return values
+
+    def _add_leave_manager_ids_in_group(self, values):
```
Comment (StephaneMangin) [inline hr_holidays_validator_list/models/hr_employee.py:29]: An onchange would have suffice to affect groups on managers.

File: hr_holidays_validator_list/models/hr_employee.py (hunk 1 above)
Comment (victorvermot) [reply hr_holidays_validator_list/models/hr_employee.py]: You're right I am going to check that

File: hr_holidays_validator_list/models/hr_employee.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+class HrEmployeeBase(models.AbstractModel):
+    _inherit = "hr.employee.base"
+
+    leave_manager_ids = fields.Many2many(
+        "res.users",
+        string="Time Off Managers",
+        store=True,
+        readonly=False,
+        help="""Select the users responsible for approving 'Time Off' of this employee.\n
```
Comment (yvaucher) [inline hr_holidays_validator_list/models/hr_employee.py]: This double the end line as `"""` will preserve the return line and the `\n` is added. Is it intended?

File: hr_holidays_validator_list/models/hr_leave_allocation.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+from odoo import models
+
+
+class HolidaysAllocation(models.Model):
+    """Allocation Requests Access specifications: similar to leave requests"""
+
+    _inherit = "hr.leave.allocation"
+
+    def activity_update(self):
+        """uUpdates activity for all leave_manager_ids"""
```
Comment (yvaucher) [inline hr_holidays_validator_list/models/hr_leave_allocation.py]:
```suggestion
        """Updates activity for all leave_manager_ids"""
```

File: hr_holidays_validator_list/models/hr_employee.py (hunk 2 above)
Comment (yvaucher) [reply hr_holidays_validator_list/models/hr_employee.py:29]: I disagree, we are on hr_employee, an onchange should only affect the values of the current form. Groups won't be available from here.

File: hr_holidays_validator_list/models/hr_employee.py (hunk 1 above)
Comment (yvaucher) [reply hr_holidays_validator_list/models/hr_employee.py]: @StephaneMangin The field `leave_manager_ids` is already a computed stored field. Question is can we insert nicely the set of the values from the M2M nicely or not. We might want to keep leave_manager_ids value to it's current natural computation.

@victorvermot 

What we could probably do to improve the code is:

Override https://github.com/odoo/odoo/blob/17.0/addons/hr_holidays/models/hr_employee.py#L152-L159
this way:

``` python
    @api.depends('parent_id', 'leave_manager_ids')
    def _compute_leave_manager(self):
            
        for employee in self:
            if self.leave_manager_ids:
               self.leave_manager_id = leave_manager_ids[0]     
           else:
                super()._compute_leave_manager()
```

**Review by yvaucher — CHANGES_REQUESTED** [review]
Small typos and a proposal of improvement of the code.

**Review by yvaucher — APPROVED** [review]

**Review by cyrilmanuel — APPROVED** [review]
lgtm

**Comment by gurneyalex** [comment]
@victorvermot you should check the CI: the tests are failing.

**Pull request closed without merging**